app/controllers/components/course/videos_component.rb

```diff
@@ -42,4 +43,14 @@ def settings_sidebar_items
       }
     ]
   end
+
+  def unread_count
+    return 0 if current_course_user.staff?
+
+    Course::Video.
+      from_course(current_course).
+      unwatched_by(current_user).
+      select(&:published?).
```

Comment:
Just to check, will this generate the SQL `where lesson_plan_item.published = true`? If yes, 👍, if not, we should have it in the same SQL query rather than rely on Ruby/AR to do this.

Reply:
I think it'll filter in Rails, let me add another scope.